File: go-version/leetcode/LC2016/maximum_difference.go
package LC2016

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/6/16 上午8:15
 * @FilePath: maximum_difference
 * @Description: 前缀最小值
 */

func maximumDifference(nums []int) int {
	ans, preMin := -1, nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] <= preMin {
			preMin = nums[i]
		} else {
			ans = max(ans, nums[i]-preMin)
		}
	}
	return ans
}

File: python-version/LC2016/maximum_difference.py
class Solution:
    @staticmethod
    def maximumDifference(nums: list[int]) -> int:
        ans, pre_min = -1, nums[0]
        for num in nums[1:]:
            if num <= pre_min:
                pre_min = num
            else:
                ans = max(ans, num - pre_min)
        return ans
